fastlane_core/lib/fastlane_core/itunes_transporter.rb

```diff
@@ -569,14 +690,32 @@ def upload(app_id = nil, dir = nil, package_path: nil, asset_path: nil)
       password_placeholder = @jwt.nil? ? 'YourPassword' : nil
       jwt_placeholder = @jwt.nil? ? nil : 'YourJWT'
 
-      command = @transporter_executor.build_upload_command(@user, @password, actual_dir, @provider_short_name, @jwt)
-      UI.verbose(@transporter_executor.build_upload_command(@user, password_placeholder, actual_dir, @provider_short_name, jwt_placeholder))
+      # Handle AppStore Connect API
+      use_api_key = !@api_key.nil?
+      api_key_placeholder = use_api_key ? { key_id: "YourKeyID", issuer_id: "YourIssuerID" } : nil
+
+      tmp_p8_dir_path = nil
+      if use_api_key
+        tmp_p8_dir_path = Dir.mktmpdir("deliver-")
+        # Specified p8 needs to be generated to call altool
+        File.open(File.join(tmp_p8_dir_path, "AuthKey_#{@api_key[:key_id]}.p8"), "w") do |p8|
```

Comment:
Given the API key might be base64, that we will decode in such case, leading to the possibility that the content written here might contain some binary data, better write to that p8 file using binary option:


```suggestion
        File.open(File.join(tmp_p8_dir_path, "AuthKey_#{@api_key[:key_id]}.p8"), "wb") do |p8|
```